### PR TITLE
v2 forward compatibility fixes with v3

### PR DESCRIPTION
**Pull request by kaelig**
- Quote ambiguous colors so that they are recognised as both strings and color data types
- [fix] In new versions of Sass, passing a string works (previously required passing a color type)

### CODE REVIEW

**Comment by AlbertoElias** [comment]
I don't understand why in the palette, some colours are both quoted and unquoted, and others are only quoted

**Comment by kaelig** [comment]
Maybe this example will make it more clear:

```scss
$colors: (
	"black": #000000, // Output is fine (black)
	black: #000000 // Class name gets minified to c-#000
);

@each $name, $csscolor in $colors {
	.c-#{$name} {
		background-color: $csscolor;
	}
}
```

Compiles to (in compressed mode):

```css
.c-black{background-color:#000}.c-#000{background-color:#000}
```

http://sassmeister.com/gist/911c0887c1fa588737da

So what is happening when going from v2 to v3 is this:

```scss
.el {
  color: oColorsGetPaletteColor(white); // v2 (color): gets #fff in the map
  color: oColorsGetPaletteColor('white'); // v3 (string): Gets 'white' in the map
}
```

To fix the bug but still be compatible with v2, we need both not quoted (v2, colors) and quoted (v3, strings) keys.

**Comment by kaelig** [comment]
Tested with o-header in v2.5.0 and v3.0.0: it works!

**Comment by AlbertoElias** [comment]
My question was why you were quoting some and not others like pink-tint5. I understand only css colours are minified wrongly, but if in the palette map, you only have the quoted "pink-tint5¨ will `oColorsGetPaletteColour(pink-tint5)` and `oColorsSetUseCase(email, text, pink-tint5)` work?

**Comment by kaelig** [comment]
May I add that in v3, this will happen:

```scss
oColorsGetPaletteColour(pink-tint5) // works
oColorsGetPaletteColour(red) // fails
```

File: src/scss/_palette.scss [hunk 1]
```diff
@@ -19,68 +19,78 @@
...
-	orange:                #d66d06,
+	pink:                    #fff1e0, // v2
+	'pink':                  #fff1e0, // v3
+	black:                   #000000, // v2
+	'black':                 #000000, // v3
+	white:                   #ffffff, // v2
+	'white':                 #ffffff, // v3
+	blue:                    #2e6e9e, // v2
+	'blue':                  #2e6e9e, // v3
+	'dark-blue':             #275e86,
```
Comment (AlbertoElias) [inline src/scss/_palette.scss]: Here for example, you've removed the unquoted one, right?

Comment (kaelig) [inline src/scss/_palette.scss]: Because in CSS, `dark-blue` is not a color (`DarkBlue` is, though) so Sass will interpret it as a string regardless.

**Comment by AlbertoElias** [comment]
Spoke with @kaelig and pr lgtm

**Pull request merged**